### PR TITLE
Metadata field restriction: Metadata Fields UI

### PR DESCRIPTION
**Pull request by joshsadam**
## Description of changes

Allows a manager to add a restriction level to a metadata field on the Project > Metadata page.

**NOTE: This doe snot currently restrict any project members from seeing the fields.** This will be done in a future PR.

## Related issue

N/A

## Checklist

Things for the developer to confirm they've done before the PR should be accepted:

*   [ ] ~CHANGELOG.md (and UPGRADING.md if necessary) updated with information for new change.~
*   [x] Tests added (or description of how to test) for any new features.
*   [ ] ~User documentation updated for UI or technical changes.~

### CODE REVIEW

File: src/main/resources/i18n/messages.properties [hunk 1]
```diff
@@ -995,8 +995,11 @@ ProjectMetadataTemplates.fields={0} fields
 
 MetadataFields.title=Metadata Fields
 
-MetadataFieldsList.create=Create Template
-MetadataFieldsList.empty=No metadata has been added to this project.
+MetadataFieldsListManager.create=Create Template
+MetadataFieldsListMember.empty=No metadata has been added to this project.
+MetadataFieldsListManager.restrictions=Restrictions
+MetadataFieldsListManager.restrictions-help=Each Metadata Field can be set to only be viewed by certain project roles.  Manager's will be able to view all fields while the Collaborators will only be able to view the fields that are selected for that role.
+server.MetadataFieldsListManager.update=Role for {0} update to {1}
```
Comment (deepsidhu85) [inline src/main/resources/i18n/messages.properties:1002]: Changed `update` to `updated` in the actual text for the translation key

File: src/main/webapp/resources/js/pages/projects/redux/fieldsSlice.js [hunk 2]
```diff
@@ -1,5 +1,9 @@
 import { createAsyncThunk, createSlice } from "@reduxjs/toolkit";
-import { getMetadataFieldsForProject } from "../../../apis/metadata/field";
+import {
+  getMetadataFieldsForProject,
+  getMetadataRestrictions,
+  patchProjectMetadataFieldRestriction,
```
Comment (deepsidhu85) [inline src/main/webapp/resources/js/pages/projects/redux/fieldsSlice.js:5]: I think I mentioned this above but change `patchProjectMetadataFieldRestriction` to `updateProjectMetadataFieldRestriction`

File: src/main/webapp/resources/js/pages/projects/redux/fieldsSlice.js [hunk 3]
```diff
@@ -17,6 +21,36 @@ export const fetchFieldsForProject = createAsyncThunk(
   }
 );
```
Comment (deepsidhu85) [inline src/main/webapp/resources/js/pages/projects/redux/fieldsSlice.js:23]: Comments for thunks

**Review by deepsidhu85 — CHANGES_REQUESTED** [review]
Looking good! Great job. Just a few changes below :)

File: src/main/webapp/resources/js/apis/metadata/field.js [hunk 4]
```diff
@@ -36,3 +36,27 @@ export async function getMetadataFieldsForProject(projectId) {
...
+export async function getMetadataRestrictions() {
+  try {
+    const { data } = await axios.get(`${URL}/restrictions`);
+    return data;
+  } catch (e) {
+    return Promise.reject(e.response.data);
+  }
+}
+
+export async function patchProjectMetadataFieldRestriction({
```
[the remark this answers is not in the document]
Comment (joshsadam) [reply src/main/webapp/resources/js/apis/metadata/field.js]: Updated in 3635ca7

File: src/main/webapp/resources/js/components/form/InputWithOptions.jsx [hunk 5]
```diff
@@ -18,6 +18,7 @@ function isTruthy(options) {
  * @constructor
  */
 export function InputWithOptions({ item }) {
+  console.log(item);
```
[the remark this answers is not in the document]
Comment (joshsadam) [reply src/main/webapp/resources/js/components/form/InputWithOptions.jsx]: Updated in daa829a

File: src/main/java/ca/corefacility/bioinformatics/irida/ria/web/ajax/metadata/MetadataAjaxController.java [hunk 6]
```diff
@@ -119,4 +123,31 @@ public MetadataAjaxController(UIMetadataService service) {
...
+	 * Update a restriction level on a metadata field for a project
+	 *
+	 * @param projectId   Identifier for the project
+	 * @param fieldId     Identifier for the metadata field
+	 * @param projectRole New project role to set the field to
+	 * @param locale      Current users {@link Locale}
+	 * @return Message to user on the status of the update
+	 */
+	@PatchMapping("/fields/restrictions")
+	public ResponseEntity<AjaxResponse> patchProjectMetadataFieldRestriction(@RequestParam Long projectId,
```
[the remark this answers is not in the document]
Comment (joshsadam) [reply src/main/java/ca/corefacility/bioinformatics/irida/ria/web/ajax/metadata/MetadataAjaxController.java]: Updated in 98138b0

File: src/main/java/ca/corefacility/bioinformatics/irida/service/impl/sample/MetadataTemplateServiceImpl.java [hunk 7]
```diff
@@ -31,9 +35,9 @@
 public class MetadataTemplateServiceImpl extends CRUDServiceImpl<Long, MetadataTemplate>
 		implements MetadataTemplateService {
 
-	private ProjectMetadataTemplateJoinRepository pmtRepository;
-	private MetadataFieldRepository fieldRepository;
-	private MetadataRestrictionRepository metadataRestrictionRepository;
+	private final ProjectMetadataTemplateJoinRepository pmtRepository;
```
[the remark this answers is not in the document]
Comment (joshsadam) [reply src/main/java/ca/corefacility/bioinformatics/irida/service/impl/sample/MetadataTemplateServiceImpl.java:38]: Fixed in 8430c1d016789bbe1271b7455a391f7303276b50 not a mac issue LOL

File: src/main/webapp/resources/js/apis/metadata/field.js [hunk 8]
```diff
@@ -36,3 +36,27 @@ export async function getMetadataFieldsForProject(projectId) {
     return e.response.data.message;
   }
 }
+
```
[the remark this answers is not in the document]
Comment (joshsadam) [reply src/main/webapp/resources/js/apis/metadata/field.js:39]: Updated in 5186c8e

File: src/main/webapp/resources/js/pages/projects/redux/fieldsSlice.js (hunk 2 above)
Comment (joshsadam) [reply src/main/webapp/resources/js/pages/projects/redux/fieldsSlice.js]: Fixed earlier

File: src/main/resources/i18n/messages.properties (hunk 1 above)
Comment (joshsadam) [reply src/main/resources/i18n/messages.properties]: Fixed in 0e81323

File: src/main/webapp/resources/js/pages/projects/redux/fieldsSlice.js (hunk 3 above)
Comment (joshsadam) [reply src/main/webapp/resources/js/pages/projects/redux/fieldsSlice.js:23]: Updated in 85f5949

File: src/main/java/ca/corefacility/bioinformatics/irida/service/impl/sample/MetadataTemplateServiceImpl.java (hunk 7 above)
[the remark this answers is not in the document]
Comment (deepsidhu85) [reply src/main/java/ca/corefacility/bioinformatics/irida/service/impl/sample/MetadataTemplateServiceImpl.java]: Haha usually is!

**Review by deepsidhu85 — APPROVED** [review]

**Pull request merged**